synapse/handlers/message.py

```diff
@@ -792,9 +793,14 @@ def handle_new_client_event(
             EventTypes.Create,
             "",
         ):
-            room_version = event.content.get("room_version", RoomVersions.V1.identifier)
+            room_version_id = event.content.get(
+                "room_version", RoomVersions.V1.identifier
+            )
+            room_version = KNOWN_ROOM_VERSIONS[room_version_id]
         else:
-            room_version = yield self.store.get_room_version_id(event.room_id)
+            room_version = yield defer.ensureDeferred(
+                self.store.get_room_version(event.room_id)
+            )
```

Comment:
What was the point of this change?

Reply:
We need the `room_version` below, not just the ID, so I had to rejigger some code.

If you're asking why I added the `defer.ensureDeferred`: `get_room_version` is async (although `get_room_version_id` is too, so not sure how this worked beforehand).

Reply:
`get_room_version_id` has an `@cached` decorator, which means it's really a function that returns a Deferred.